Clean up native auth common logging

Yuki-YuXin opened the pull request:
### Goal:
- Review our log statements and reduce to only those that are relevant and useful. Especially the logMethodCall should be much reduced. Not every method call is relevant to know about when debugging.
- Double check that the existing logs use string interpolation correctly to avoid leaking sensitive information in the logs.

### Changes summary:
- Every log item has correlation argument. (make sure use getCorrelationId() )
- The [NativeAuthMsalController.kt](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/2374/files#diff-304daa7418210ef9cd11ee385364a4a6f74256dc627f404e53422ec50f75b282) layer acts as the main log layer. Instead, remove logSessions under OAuth2Configuration, OAuth2Strategy, RequestProvider since the users care more about the outcome of the request than the construction and sending of the request. Apply the rule "only logger unknown error" to NativeAuthMsalController.
- [Log challengeTypesWithDefault rather than passed challengeTypes](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/2374/commits/be777c5813c6c15518b358a718b863562d92f5b3)
- Complete // TODO add safe logging using string interpolation with curly brackets and check the rest according to MSAL practice under MicrosoftStsTokenResponse.
- Add request and result Logger.verbose in interactors.

### TODO

- Test PII
- infoWithObject(final String tag, final String correlationID, final String message, final ILoggable object) like warnWithObject
- every log has correlationID


### Company PR:
https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/2071

Yuki-YuXin committed:
Remove LogSession and Logger(result) in all commands

Yuki-YuXin committed:
1. getCorrelationId() -> correlationId
2. Add logs in order to remove networking layer's logging
3. Apply the rule: only log unknown error

Yuki-YuXin committed:
Remove all logSessions under NativeAuthOAuth2Strategy

Yuki-YuXin committed:
Remove all logSessions under Interactors

Yuki-YuXin committed:
Remove all logSessions under RequestProvider

Yuki-YuXin committed:
Remove all logSession under NativeAuthOAuth2Configuration

Yuki-YuXin committed:
Remove logging in all responses.

Yuki-YuXin committed:
Log challengeTypesWithDefault rather than passed challengeTypes

Yuki-YuXin committed:
Complete // TODO add safe logging using String.format()

Yuki-YuXin committed:
Checking: Use the curly brackets around the class members we intended to log

SammyO commented on common/src/main/java/com/microsoft/identity/common/nativeauth/internal/commands/AcquireTokenNoFixedScopesCommand.kt:
```diff
@@ -47,20 +45,9 @@ class AcquireTokenNoFixedScopesCommand(
             parameters
         )
-
-        Logger.info(
```
Is the result logged in MSAL's SDK interface (when receiving the result) or in the Controller (when returning the result)? In my opinion this is valuable information. it's the summary of all the internal processing the SDK does.

SammyO replied:
If we decide to keep this, we should probably add the correlation ID.

Yuki-YuXin replied:
The command layer is ignored in the first version. Put them back and add correlation ID for the Logger.

SammyO commented on common/src/main/java/com/microsoft/identity/common/nativeauth/internal/controllers/NativeAuthMsalController.kt:
```diff
@@ -126,7 +126,7 @@ class NativeAuthMsalController : BaseNativeAuthController() {
         LogSession.logMethodCall(
             tag = TAG,
-            correlationId = parameters.getCorrelationId(),
+            correlationId = parameters.correlationId,
```
I don't think this compiles/works. See comment above `public String getCorrelationId()` in `CommandParameters.java`

SammyO replied:
Which raises the question: did you test this? I.e. validate that correlation IDs are correctly logged?

Yuki-YuXin replied:
It somehow works in my local.
![image](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/assets/111055878/b04ba7f0-83c1-459a-ad33-96a5bdd3d575)

Just inc case, have reverted the changes to use getCorrelationId() but haven't check the consistent of correlation IDs of the flow.

Yuki-YuXin replied:
Has validated that correlation IDs are correctly logged.

SammyO commented on common4j/src/main/com/microsoft/identity/common/java/nativeauth/providers/NativeAuthResponseHandler.kt:
```diff
@@ -385,7 +385,12 @@ class NativeAuthResponseHandler {
             )
-            // TODO add safe logging
+
+            Logger.info(TAG, "MicrosoftStsTokenResponse authority:${apiResponse.authority} " +
```
Why are we logging this at all? Does MSAL log it?

Yuki-YuXin replied:
Here's our previous log and TODO. In my opinion, we don't have to log them. MSAL has authority and (request/granted)  scopes hint but in different place. 

In terms of the TokenResponse of MSAL:
MicrosoftStsOAuth2Strategy:TokenResult:MicrosoftStsTokenResponse INFO {"ext_expires_in":4165,"expires_in":4165,"mResponseReceivedTime":0,"scope":"email openid profile Application.Read.All User.Read","token_type":"Bearer"}

BaseController:TokenResult:MicrosoftStsTokenResponse INFO {"ext_expires_in":4165,"mRefreshTokenAge":"","mSpeRing":"","expires_in":4165,"mResponseReceivedTime":0,"scope":"email openid profile Application.Read.All User.Read","token_type":"Bearer"}

Thus update the contents of the log to include what MSAL displays.

SammyO commented on common4j/src/main/com/microsoft/identity/common/java/nativeauth/providers/interactors/ResetPasswordInteractor.kt:
```diff
@@ -62,12 +61,6 @@ class ResetPasswordInteractor(
     fun performResetPasswordStart(
         parameters: ResetPasswordStartCommandParameters
     ): ResetPasswordStartApiResult {
-        LogSession.logMethodCall(
```
I agree to remove the logs from OAuth2Strategy, RequestProvider and RequestHandler, but wouldn't it be useful to have some logging in the API layer? Since in the controller we're only logging method calls (not the paths through the business logic that's taken), having API-layer logging will help us figure out what API calls were made.

Yuki-YuXin replied:
Agree that API-layer need some logging for us to debug. Put logSession back to the interactor and plan to add request and response logging of the API-layer.

SammyO commented on common4j/src/main/com/microsoft/identity/common/java/nativeauth/providers/interactors/SignUpInteractor.kt:
```diff
@@ -60,18 +59,12 @@ class SignUpInteractor(
         try {
             return performSignUpStart(
-                requestCorrelationId = commandParameters.getCorrelationId(),
+                requestCorrelationId = commandParameters.correlationId,
```
Same here. Please check throughout the PR.

SammyO commented on common4j/src/main/com/microsoft/identity/common/java/nativeauth/providers/responses/resetpassword/ResetPasswordContinueApiResponse.kt:
```diff
@@ -57,11 +56,6 @@ class ResetPasswordContinueApiResponse(
      * @see com.microsoft.identity.common.java.nativeauth.providers.responses.resetpassword.ResetPasswordContinueApiResult
      */
     fun toResult(): ResetPasswordContinueApiResult {
-        LogSession.logMethodCall(
```
Since we're doing quite some API response validation in these `toResult()` methods, I'm wondering whether it's useful to log the result of that. Perhaps that logging can be done in the Interactor; a log through which we can determine how the SDK has parsed the API response and what ApiResult was generated accordingly.

Yuki-YuXin replied:
Put rawResponse -> apiResult logging in interactors.

Yuki-YuXin committed:
Revert "Remove LogSession and Logger(result) in all commands"

This reverts commit 614eea5b12edd20bc1be26edd42b2e30ae1bdc73.

Yuki-YuXin committed:
Revert "1. getCorrelationId() -> correlationId"

This reverts commit 57eb262cae3197d320e42aa1aabdf45c3c3182b7.

Yuki-YuXin committed:
Add parameters.getCorrelationId() to the Logger in commands.

Yuki-YuXin committed:
Check parameters.parameters.getCorrelationId() back to parameters.correlationId

Yuki-YuXin committed:
Interactor logging:
1) The public function logSession uses methodName=fun(argument), while the private function uses only the bare name.
2) Logger the request.

Yuki-YuXin committed:
Interactor logging:
3) Add rawApiResponse -> result logging
4) User VERBOSE instead of INFO.
5) Add correlationId into every logging.

Yuki-YuXin committed:
Add LogSession for performXXX in NativeAuthMsalController.kt

Yuki-YuXin committed:
Update MicrosoftStsTokenResponse log connect under NativeAuthResponseHandler and remove mistake import under SignInInteractor

rmccahill commented on common4j/src/main/com/microsoft/identity/common/java/nativeauth/providers/interactors/SignInInteractor.kt:
```diff
@@ -61,12 +62,20 @@ class SignInInteractor(
+        Logger.verbose(
+            "${TAG}.performSignInInitiate",
+            parameters.getCorrelationId(),
+            "performSignInInitiate: request = $request"
```
Should we be logging the full requests + responses? Could they contain potentially sensitive information like sent passwords, codes, etc?

SammyO replied:
Excellent point @rmccahill, and coincidentally: https://identitydivision.visualstudio.com/Engineering/_boards/board/t/DevExDub%20-B2C/Backlog%20items/?workitem=2900824

@Yuki-YuXin let's hold off with this PR for now. I need to work together with you to ensure we're being even more careful with logging.
